test/txsim/account.go

```diff
@@ -224,15 +226,43 @@ func (am *AccountManager) Submit(ctx context.Context, op Operation) error {
 	}
 
 	opts := make([]user.TxOption, 0)
-	if op.GasLimit == 0 {
-		opts = append(opts, user.SetGasLimit(DefaultGasLimit), user.SetFee(defaultFee))
-	} else {
-		opts = append(opts, user.SetGasLimit(op.GasLimit))
-		if op.GasPrice > 0 {
-			opts = append(opts, user.SetFee(uint64(math.Ceil(float64(op.GasLimit)*op.GasPrice))))
-		} else {
-			opts = append(opts, user.SetFee(uint64(math.Ceil(float64(op.GasLimit)*appconsts.DefaultMinGasPrice))))
+	var gasLimit, fee uint64
+	var gasPrice float64
+
+	// Determine gas limit
+	switch {
+	case op.GasLimit > 0:
+		gasLimit = op.GasLimit
+	case am.gasLimit > 0:
+		gasLimit = am.gasLimit
+	default:
+		gasLimit = DefaultGasLimit
+		fee = defaultFee
```

Comment:
[question] I'm a little confused by why this line is here. 
```suggestion
```

can this code set fee _after_ determining the gas limit and gas price?

Reply:
but gave it a shot https://github.com/celestiaorg/celestia-app/pull/4447/commits/a71af690d82df86406738487af31a3b538ea0882